pygmt/datasets/samples.py

```diff
@@ -1,13 +1,67 @@
 """
 Functions to load sample data.
 """
+import warnings
+
 import pandas as pd
+from pygmt.exceptions import GMTInvalidInput
 from pygmt.src import which
 
 
+def list_sample_dataframes():
+    """
+    Report tabular datasets available for tests and documentation examples.
+
+    Returns
+    -------
+    output : dict
+        Names and short descriptions of available sample dataframes.
+    """
+    names = {
+        "japan_quakes": "Table of earthquakes around Japan from NOAA NGDC database",
+        "ocean_ridge_points": "Table of ocean ridge points for the entire world",
+    }
+    return names
+
+
+def load_sample_dataframe(name):
+    """
+    Load an example dataset from the GMT server.
+
+    The data are downloaded to a cache directory (usually ``~/.gmt/cache``) the
+    first time you invoke this function. Afterwards, it will load the data from
+    the cache. So you'll need an internet connection the first time around.
+
+    Parameters
+    ----------
+    name : str
+        Name of the dataset to load.
+
+    Returns
+    -------
+    output : pandas.DataFrame
+        Tabular dataset.
+    """
+    names = list_sample_dataframes()
+    if name not in names:
+        raise GMTInvalidInput(f"Invalid dataset name '{name}'")
+
+    if name == "japan_quakes":
+        data = load_tut_quakes()
+
+    if name == "ocean_ridge_points":
+        data = load_ridge()
+
+    return data
+
+
 def load_japan_quakes():
     """
-    Load a table of earthquakes around Japan as a pandas.DataFrame.
+    Load a table of earthquakes around Japan as a pandas.DataFrame (Deprecated)
+
+    .. warning:: Deprecated since v0.6.0. This function has been replaced with
+       ``load_sample_dataframe(name="tut_quakes.ngdc")`` and will be removed in
+       v0.9.0.
```

Comment:
Shouldn't this ve removed in v0.8.0, i.e. 2 major versions? Same with the other ones below.
```suggestion
       v0.8.0.
```

Reply:
I thought deprecating a whole function is a more major change than the alias updates, so I went on the longer side of the 2-4 releases recommendation in the contributing guide. I'm open to a shorter deprecation cycle though.

Reply:
I'm in favor of 3 releases.

Reply:
Ok, stick to 3 minor versions then (i.e. v0.9.0).